cfme/tests/automate/test_method.py

```diff
@@ -60,6 +60,9 @@ def test_method_crud(klass):
         method.name = fauxfactory.gen_alphanumeric(8)
         method.script = "bar"
     assert method.exists
+    navigate_to(appliance.server, 'Dashboard')
```

Comment:
doesn't this mean that is_displayed is incorrect for this page ?

Reply:
@ganeshhubale thanks for the thorough descriptions of your findings, that helps a lot!

I think that this is ultimately just a workaround for the behavior you are seeing, you haven't fixed anything in the framework, just addressed it in this one test. If any other test edits a method twice, it will run into the same behavior.

I think the real problem here lies with both the implementation of `Method.update` (snippet below) and the implementation of `MethodEditView.is_displayed`
```python
if changed:
            view.save_button.click()
        else:
            view.cancel_button.click()
        view = self.create_view(MethodDetailsView, override=updates)
        view.wait_displayed()
```

Here we are assuming the view that MIQ lands on after the save/cancel button is clicked. This assumption is wrong and needs to be corrected.

Second assumption we're making is in `Method.is_displayed`, where we assume how the page was navigated to by checking the accordion tree.  MIQ does not behave that way, and will not always have the method selected in the accordion.  It could be argued that this is actually an MIQ bug, and we could write a BZ that would likely be treated as an RFE.

Thoughts:
1. The flash message assertions in `Method.update` should not be there, they should be in the `Method` crud test itself. (tangentially related)
2. `Method.update` should only use the view to call `view.flash_messages.assert_no_error`. It should not actually assert the view is displayed, or it should allow for one or the other to be displayed. The flash message assertions could be completed by creating a view for `BaseLoggedInPage`
3. There are other Entities and Views in the framework that handle this by having multiple navigation methods - 'EditFromDetails' kind of destinations, that specify the navigation path - via collection, via dashboard, via details page, etc. This is basically the source of both problems - assumptions on both ends of how the page was navigated to.

So I think we would require two fixes here, and have one optional enhancement:
Fix1: Update is_displayed to not rely on the accordion, use other page elements, title and partial accordion may be enough
Fix2: Update `Method.update` to just use `BaseLoggedInPage` for flash message assertion, don't assert the view itself is displayed. Alternatively check for any of the expected views being displayed
Optional1: Move flash message string checks to `test_method_crud`, only `assert_no_error` in `update` itself.